### PR TITLE
doc: add a minimal description for `make deploy`

### PR DESCRIPTION
**Pull request by nixpanic**
Deploying with `make deploy` is the easiest while testing and
developing. This is the recommended method for now.

Updates: #14

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -48,5 +48,46 @@ By listing the `CSIAddonsNode` CRs, the CSI-Addons Controller knows how to
...
 
+### Installation
+
+This project uses `kustomize` and a `Makefile` for deploying. By running the
+command
+
+```console
+$ make deploy
+...
+customresourcedefinition.apiextensions.k8s.io/csiaddonsnodes.csiaddons.openshift.io created
```
Comment (Madhu-1) [inline README.md:59]: @nixpanic looks like fencing CRD is missing here

Comment (nixpanic) [reply README.md:59]: Indeed, but it is just a (partial) copy/paste of what happens now. It seems that `config/crd/bases/csiaddons.openshift.io_networkfences.yaml` is not applied automatically. This should be addressed in a different PR.

File: Makefile [hunk 2]
```diff
@@ -1,6 +1,6 @@
 
 # Image URL to use all building/pushing image targets
-IMG ?= controller:latest
+IMG ?= quay.io/csiaddons/k8s-controller:latest
```
Comment (Madhu-1) [inline Makefile:3]: the image name in deployment artifacts need to be updated with new name

Comment (nixpanic) [reply Makefile:3]: where? `make deploy` seems to work with this change, but not with the original name

File: README.md (hunk 1 above)
Comment (nixpanic) [reply README.md:59]: reported this as #50.

File: Makefile (hunk 2 above)
Comment (Madhu-1) [reply Makefile:3]: okay `make deploy` edit the image name before applying

**Review by Madhu-1 — APPROVED** [review]

**Review by Rakshith-R — APPROVED** [review]

**Pull request merged**